File: lib/schwi/commands/redis.ex
defmodule Schwi.Commands.Redis do
  def run(msg, args) do
    joined_args = Enum.join(args, " ")

    if String.trim(joined_args) == "" do
      Schwi.Lib.Message.error(msg, "Provide a redis command to run.")
    else
      trimmed_args = Enum.map(args, fn arg -> String.trim(arg) end)
      response = Redix.command(:redix, trimmed_args)

      case response do
        {:ok, response} ->
          Schwi.Lib.Message.success(msg, response)

        {:error, %Redix.Error{message: message}} ->
          Schwi.Lib.Message.error(msg, message)
      end
    end
  end
end

File: src/commands/Redis.ts
import Command from "eris-boreas/lib/src/conversation/Command";
import Conversation from "eris-boreas/lib/src/conversation/Conversation";
import { success } from "../lib/transformer";

export default class Reds implements Command {
  public aliases = ["redis", "set", "get", "del"];
  public description = "Send a message to Redis";
  public usage = "redis <command> <key> <value>";

  public async run(conversation: Conversation, args: string[]) {
    if (args[0].toLowerCase() == "redis") {
      args.shift();
    }
    const command = args.shift() || "";
    const redis = conversation.eris.redis;

    if (command.toLowerCase() == "set") {
      //@ts-ignore
      return success(await redis.set(args.shift(), args.join(" ")));
    } else if (command.toLowerCase() == "get") {
      //@ts-ignore
      return success(await redis.get(args.join(" ")));
    } else if (command.toLowerCase() == "del") {
      //@ts-ignore
      return success((await redis.del(args.join(" "))).toString());
    } else {
      throw new Error("Invalid redis command");
      return ""; // never reached but typescript complains
    }
  }
}
